vendors/espressif/boards/esp32/CMakeLists.txt

```diff
@@ -42,6 +42,8 @@ target_compile_options(
     $<$<COMPILE_LANGUAGE:CXX>:${IDF_CXX_COMPILE_OPTIONS}>
 )
 
+add_compile_options(-Wno-unused-variable -Wno-unused-function -Wno-unused-function -Wno-unused-variable -Wno-unused-but-set-variable -Wno-clobbered)
```

Comment:
This isn't fixing any warnings, it is just disabling them.  Why can't these warnings be eliminated by fixing the problems in the source code?

Reply:
> This isn't fixing any warnings, it is just disabling them. Why can't these warnings be eliminated by fixing the problems in the source code?


These warnings are related to Amazon FreeRTOS and not ESP32 port. Due to strict compilation checks, these warnings are observed. Hence, we have disabled the warnings. I have attached logs for your reference.

```In file included from ../libraries/freertos_plus/aws/ota/src/mqtt/aws_iot_ota_mqtt.c:38:0:
../libraries/freertos_plus/aws/ota/src/mqtt/aws_iot_ota_mqtt.c: In function 'prvSendCallbackEvent':
../libraries/freertos_plus/aws/ota/include/aws_iot_ota_agent.h:56:65: warning: unused variable 'OTA_METHOD_NAME' [-Wunused-variable]
     #define DEFINE_OTA_METHOD_NAME( name )    static const char OTA_METHOD_NAME[] = name;
                                                                 ^
../libraries/freertos_plus/aws/ota/src/mqtt/aws_iot_ota_mqtt.c:370:5: note: in expansion of macro 'DEFINE_OTA_METHOD_NAME'
     DEFINE_OTA_METHOD_NAME( "prvSendCallbackEvent" );
     ^
../libraries/freertos_plus/aws/ota/src/mqtt/aws_iot_ota_mqtt.c: In function 'prvJobPublishCallback':
../libraries/freertos_plus/aws/ota/include/aws_iot_ota_agent.h:56:65: warning: unused variable 'OTA_METHOD_NAME' [-Wunused-variable]
     #define DEFINE_OTA_METHOD_NAME( name )    static const char OTA_METHOD_NAME[] = name;
                                                                 ^
../libraries/freertos_plus/aws/ota/src/mqtt/aws_iot_ota_mqtt.c:421:5: note: in expansion of macro 'DEFINE_OTA_METHOD_NAME'
     DEFINE_OTA_METHOD_NAME( "prvJobPublishCallback" );
     ^
../libraries/freertos_plus/aws/ota/src/mqtt/aws_iot_ota_mqtt.c: In function 'prvDataPublishCallback':
../libraries/freertos_plus/aws/ota/include/aws_iot_ota_agent.h:56:65: warning: unused variable 'OTA_METHOD_NAME' [-Wunused-variable]
     #define DEFINE_OTA_METHOD_NAME( name )    static const char OTA_METHOD_NAME[] = name;
                                                                 ^
../libraries/freertos_plus/aws/ota/src/mqtt/aws_iot_ota_mqtt.c:437:5: note: in expansion of macro 'DEFINE_OTA_METHOD_NAME'
     DEFINE_OTA_METHOD_NAME( "prvDataPublishCallback" );
     ^
../libraries/freertos_plus/aws/ota/src/mqtt/aws_iot_ota_mqtt.c: In function 'prvDecodeFileBlock_Mqtt':
../libraries/freertos_plus/aws/ota/include/aws_iot_ota_agent.h:56:65: warning: unused variable 'OTA_METHOD_NAME' [-Wunused-variable]
     #define DEFINE_OTA_METHOD_NAME( name )    static const char OTA_METHOD_NAME[] = name;
                                                                 ^
../libraries/freertos_plus/aws/ota/src/mqtt/aws_iot_ota_mqtt.c:850:5: note: in expansion of macro 'DEFINE_OTA_METHOD_NAME'
     DEFINE_OTA_METHOD_NAME( "prvDecodeFileBlock_Mqtt" );
     ^
../libraries/freertos_plus/aws/ota/src/mqtt/aws_iot_ota_mqtt.c: In function 'prvCleanup_Mqtt':
../libraries/freertos_plus/aws/ota/include/aws_iot_ota_agent.h:56:65: warning: unused variable 'OTA_METHOD_NAME' [-Wunused-variable]
     #define DEFINE_OTA_METHOD_NAME( name )    static const char OTA_METHOD_NAME[] = name;
                                                                 ^
../libraries/freertos_plus/aws/ota/src/mqtt/aws_iot_ota_mqtt.c:886:5: note: in expansion of macro 'DEFINE_OTA_METHOD_NAME'
     DEFINE_OTA_METHOD_NAME( "prvCleanup_Mqtt" );
     ^
[30/899] Building C object CMakeFiles/afr_https.dir/libraries/c_sdk/standard/https/src/iot_https_utils.c.obj
In file included from ../libraries/c_sdk/standard/https/src/iot_https_utils.c:37:0:
../libraries/c_sdk/standard/https/src/private/iot_https_internal.h:490:21: warning: '_pHttpsMethodStrings' defined but not used [-Wunused-variable]
 static const char * _pHttpsMethodStrings[] = {
                     ^
[32/899] Building C object CMakeFiles/afr_ota.dir/libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c.obj
In file included from ../libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c:40:0:
../libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c: In function 'prvSetImageStateWithReason':
../libraries/freertos_plus/aws/ota/include/aws_iot_ota_agent.h:56:65: warning: unused variable 'OTA_METHOD_NAME' [-Wunused-variable]
     #define DEFINE_OTA_METHOD_NAME( name )    static const char OTA_METHOD_NAME[] = name;
                                                                 ^
../libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c:519:5: note: in expansion of macro 'DEFINE_OTA_METHOD_NAME'
     DEFINE_OTA_METHOD_NAME( "prvSetImageStateWithReason" );
     ^
../libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c: In function 'prvStartHandler':
../libraries/freertos_plus/aws/ota/include/aws_iot_ota_agent.h:56:65: warning: unused variable 'OTA_METHOD_NAME' [-Wunused-variable]
     #define DEFINE_OTA_METHOD_NAME( name )    static const char OTA_METHOD_NAME[] = name;
                                                                 ^
../libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c:723:5: note: in expansion of macro 'DEFINE_OTA_METHOD_NAME'
     DEFINE_OTA_METHOD_NAME( "prvStartHandler" );
     ^
../libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c: In function 'prvRequestJobHandler':
../libraries/freertos_plus/aws/ota/include/aws_iot_ota_agent.h:56:65: warning: unused variable 'OTA_METHOD_NAME' [-Wunused-variable]
     #define DEFINE_OTA_METHOD_NAME( name )    static const char OTA_METHOD_NAME[] = name;
                                                                 ^
../libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c:770:5: note: in expansion of macro 'DEFINE_OTA_METHOD_NAME'
     DEFINE_OTA_METHOD_NAME( "prvRequestJobHandler" );
     ^
../libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c: In function 'prvCloseFileHandler':
../libraries/freertos_plus/aws/ota/include/aws_iot_ota_agent.h:56:65: warning: unused variable 'OTA_METHOD_NAME' [-Wunused-variable]
     #define DEFINE_OTA_METHOD_NAME( name )    static const char OTA_METHOD_NAME[] = name;
                                                                 ^
../libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c:1125:5: note: in expansion of macro 'DEFINE_OTA_METHOD_NAME'
     DEFINE_OTA_METHOD_NAME( "prvCloseFileHandler" );
     ^
../libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c: In function 'OTA_AgentInit':
../libraries/freertos_plus/aws/ota/include/aws_iot_ota_agent.h:56:65: warning: unused variable 'OTA_METHOD_NAME' [-Wunused-variable]
     #define DEFINE_OTA_METHOD_NAME( name )    static const char OTA_METHOD_NAME[] = name;
                                                                 ^
../libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c:2496:5: note: in expansion of macro 'DEFINE_OTA_METHOD_NAME'
     DEFINE_OTA_METHOD_NAME( "OTA_AgentInit" );
     ^
../libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c: At top level:
../libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c:1087:18: warning: 'prvUserAbortHandler' defined but not used [-Wunused-function]
 static OTA_Err_t prvUserAbortHandler( OTA_EventData_t * pxEventData )
                  ^
../libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c:1138:18: warning: 'prvErrorHandler' defined but not used [-Wunused-function]
 static OTA_Err_t prvErrorHandler( OTA_EventData_t * pxEventData )
                  ^
../libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c:1148:18: warning: 'prvIdleHandler' defined but not used [-Wunused-function]
 static OTA_Err_t prvIdleHandler( OTA_EventData_t * pxEventData )
                  ^
[38/899] Building C object CMakeFiles/afr_mqtt.dir/libraries/c_sdk/standard/mqtt/src/iot_mqtt_serialize.c.obj
In file included from ../libraries/c_sdk/standard/common/include/iot_atomic.h:37:0,
                 from ../libraries/c_sdk/standard/mqtt/src/iot_mqtt_serialize.c:47:
../freertos_kernel/include/atomic.h:104:34: warning: 'Atomic_CompareAndSwap_u32' defined but not used [-Wunused-function]
 static portFORCE_INLINE uint32_t Atomic_CompareAndSwap_u32(
                                  ^
../freertos_kernel/include/atomic.h:138:32: warning: 'Atomic_SwapPointers_p32' defined but not used [-Wunused-function]
 static portFORCE_INLINE void * Atomic_SwapPointers_p32(
                                ^
../freertos_kernel/include/atomic.h:171:34: warning: 'Atomic_CompareAndSwapPointers_p32' defined but not used [-Wunused-function]
 static portFORCE_INLINE uint32_t Atomic_CompareAndSwapPointers_p32(
                                  ^
../freertos_kernel/include/atomic.h:233:34: warning: 'Atomic_Subtract_u32' defined but not used [-Wunused-function]
 static portFORCE_INLINE uint32_t Atomic_Subtract_u32(
                                  ^
../freertos_kernel/include/atomic.h:260:34: warning: 'Atomic_Increment_u32' defined but not used [-Wunused-function]
 static portFORCE_INLINE uint32_t Atomic_Increment_u32( uint32_t volatile * pAddend )
                                  ^
../freertos_kernel/include/atomic.h:285:34: warning: 'Atomic_Decrement_u32' defined but not used [-Wunused-function]
 static portFORCE_INLINE uint32_t Atomic_Decrement_u32( uint32_t volatile * pAddend )
                                  ^
../freertos_kernel/include/atomic.h:313:34: warning: 'Atomic_OR_u32' defined but not used [-Wunused-function]
 static portFORCE_INLINE uint32_t Atomic_OR_u32(
                                  ^
../freertos_kernel/include/atomic.h:341:34: warning: 'Atomic_AND_u32' defined but not used [-Wunused-function]
 static portFORCE_INLINE uint32_t Atomic_AND_u32(
                                  ^
../freertos_kernel/include/atomic.h:369:34: warning: 'Atomic_NAND_u32' defined but not used [-Wunused-function]
 static portFORCE_INLINE uint32_t Atomic_NAND_u32(
                                  ^
../freertos_kernel/include/atomic.h:397:34: warning: 'Atomic_XOR_u32' defined but not used [-Wunused-function]
 static portFORCE_INLINE uint32_t Atomic_XOR_u32(
                                  ^
[52/899] Building C object CMakeFiles/afr_defender.dir/libraries/c_sdk/aws/defender/src/aws_iot_defender_api.c.obj
../libraries/c_sdk/aws/defender/src/aws_iot_defender_api.c: In function '_metricsPublishRoutine':
../libraries/c_sdk/aws/defender/src/aws_iot_defender_api.c:466:24: warning: variable 'taskPoolError' set but not used [-Wunused-but-set-variable]
     IotTaskPoolError_t taskPoolError = IOT_TASKPOOL_SUCCESS;
                        ^
[72/899] Building C object CMakeFiles/afr_ble.dir/libraries/c_sdk/standard/ble/src/services/device_information/iot_ble_device_information.c.obj
../libraries/c_sdk/standard/ble/src/services/device_information/iot_ble_device_information.c: In function '_deviceInfoMTUCharCallback':
../libraries/c_sdk/standard/ble/src/services/device_information/iot_ble_device_information.c:386:27: warning: unused variable 'resp' [-Wunused-variable]
     IotBleEventResponse_t resp;
                           ^
../libraries/c_sdk/standard/ble/src/services/device_information/iot_ble_device_information.c:385:27: warning: unused variable 'attrData' [-Wunused-variable]
     IotBleAttributeData_t attrData = { 0 };
                           ^
[88/899] Building C object CMakeFiles/afr_ble.dir/libraries/c_sdk/standard/ble/src/services/mqtt_ble/iot_ble_data_transfer.c.obj
../libraries/c_sdk/standard/ble/src/services/mqtt_ble/iot_ble_data_transfer.c: In function '_ControlCharCallback':
../libraries/c_sdk/standard/ble/src/services/mqtt_ble/iot_ble_data_transfer.c:523:13: warning: unused variable 'value' [-Wunused-variable]
     uint8_t value;
             ^
../libraries/c_sdk/standard/ble/src/services/mqtt_ble/iot_ble_data_transfer.c: In function '_clientCharCfgDescrCallback':
../libraries/c_sdk/standard/ble/src/services/mqtt_ble/iot_ble_data_transfer.c:784:14: warning: unused variable 'CCFGValue' [-Wunused-variable]
     uint16_t CCFGValue;
              ^
../libraries/c_sdk/standard/ble/src/services/mqtt_ble/iot_ble_data_transfer.c: In function '_connectionCallback':
../libraries/c_sdk/standard/ble/src/services/mqtt_ble/iot_ble_data_transfer.c:831:35: warning: unused variable 'pService' [-Wunused-variable]
     IotBleDataTransferService_t * pService;
                                   ^
../libraries/c_sdk/standard/ble/src/services/mqtt_ble/iot_ble_data_transfer.c: In function 'IotBleDataTransfer_Init':
../libraries/c_sdk/standard/ble/src/services/mqtt_ble/iot_ble_data_transfer.c:950:35: warning: unused variable 'pService' [-Wunused-variable]
     IotBleDataTransferService_t * pService = NULL;
                                   ^
../libraries/c_sdk/standard/ble/src/services/mqtt_ble/iot_ble_data_transfer.c:949:17: warning: unused variable 'callbacksRegistered' [-Wunused-variable]
     static bool callbacksRegistered = false;
                 ^
../libraries/c_sdk/standard/ble/src/services/mqtt_ble/iot_ble_data_transfer.c: In function 'IotBleDataTransfer_Send':
../libraries/c_sdk/standard/ble/src/services/mqtt_ble/iot_ble_data_transfer.c:1116:15: warning: unused variable 'pData' [-Wunused-variable]
     uint8_t * pData;
               ^
[146/899] Building C object CMakeFiles/afr_common.dir/libraries/c_sdk/standard/common/iot_init.c.obj
In file included from ../libraries/c_sdk/standard/common/include/iot_atomic.h:37:0,
                 from ../libraries/c_sdk/standard/common/iot_init.c:41:
../freertos_kernel/include/atomic.h:104:34: warning: 'Atomic_CompareAndSwap_u32' defined but not used [-Wunused-function]
 static portFORCE_INLINE uint32_t Atomic_CompareAndSwap_u32(
                                  ^
../freertos_kernel/include/atomic.h:138:32: warning: 'Atomic_SwapPointers_p32' defined but not used [-Wunused-function]
 static portFORCE_INLINE void * Atomic_SwapPointers_p32(
                                ^
../freertos_kernel/include/atomic.h:171:34: warning: 'Atomic_CompareAndSwapPointers_p32' defined but not used [-Wunused-function]
 static portFORCE_INLINE uint32_t Atomic_CompareAndSwapPointers_p32(
                                  ^
../freertos_kernel/include/atomic.h:204:34: warning: 'Atomic_Add_u32' defined but not used [-Wunused-function]
 static portFORCE_INLINE uint32_t Atomic_Add_u32(
                                  ^
../freertos_kernel/include/atomic.h:233:34: warning: 'Atomic_Subtract_u32' defined but not used [-Wunused-function]
 static portFORCE_INLINE uint32_t Atomic_Subtract_u32(
                                  ^
../freertos_kernel/include/atomic.h:260:34: warning: 'Atomic_Increment_u32' defined but not used [-Wunused-function]
 static portFORCE_INLINE uint32_t Atomic_Increment_u32( uint32_t volatile * pAddend )
                                  ^
../freertos_kernel/include/atomic.h:285:34: warning: 'Atomic_Decrement_u32' defined but not used [-Wunused-function]
 static portFORCE_INLINE uint32_t Atomic_Decrement_u32( uint32_t volatile * pAddend )
                                  ^
../freertos_kernel/include/atomic.h:313:34: warning: 'Atomic_OR_u32' defined but not used [-Wunused-function]
 static portFORCE_INLINE uint32_t Atomic_OR_u32(
                                  ^
../freertos_kernel/include/atomic.h:341:34: warning: 'Atomic_AND_u32' defined but not used [-Wunused-function]
 static portFORCE_INLINE uint32_t Atomic_AND_u32(
                                  ^
../freertos_kernel/include/atomic.h:369:34: warning: 'Atomic_NAND_u32' defined but not used [-Wunused-function]
 static portFORCE_INLINE uint32_t Atomic_NAND_u32(
                                  ^
../freertos_kernel/include/atomic.h:397:34: warning: 'Atomic_XOR_u32' defined but not used [-Wunused-function]
 static portFORCE_INLINE uint32_t Atomic_XOR_u32(
                                  ^
[150/899] Building C object CMakeFiles/afr_common.dir/libraries/c_sdk/standard/common/iot_device_metrics.c.obj
In file included from ../libraries/c_sdk/standard/common/include/iot_atomic.h:37:0,
                 from ../libraries/c_sdk/standard/common/iot_device_metrics.c:40:
../freertos_kernel/include/atomic.h:138:32: warning: 'Atomic_SwapPointers_p32' defined but not used [-Wunused-function]
 static portFORCE_INLINE void * Atomic_SwapPointers_p32(
                                ^
../freertos_kernel/include/atomic.h:171:34: warning: 'Atomic_CompareAndSwapPointers_p32' defined but not used [-Wunused-function]
 static portFORCE_INLINE uint32_t Atomic_CompareAndSwapPointers_p32(
                                  ^
../freertos_kernel/include/atomic.h:204:34: warning: 'Atomic_Add_u32' defined but not used [-Wunused-function]
 static portFORCE_INLINE uint32_t Atomic_Add_u32(
                                  ^
../freertos_kernel/include/atomic.h:233:34: warning: 'Atomic_Subtract_u32' defined but not used [-Wunused-function]
 static portFORCE_INLINE uint32_t Atomic_Subtract_u32(
                                  ^
../freertos_kernel/include/atomic.h:260:34: warning: 'Atomic_Increment_u32' defined but not used [-Wunused-function]
 static portFORCE_INLINE uint32_t Atomic_Increment_u32( uint32_t volatile * pAddend )
                                  ^
../freertos_kernel/include/atomic.h:285:34: warning: 'Atomic_Decrement_u32' defined but not used [-Wunused-function]
 static portFORCE_INLINE uint32_t Atomic_Decrement_u32( uint32_t volatile * pAddend )
                                  ^
../freertos_kernel/include/atomic.h:313:34: warning: 'Atomic_OR_u32' defined but not used [-Wunused-function]
 static portFORCE_INLINE uint32_t Atomic_OR_u32(
                                  ^
../freertos_kernel/include/atomic.h:341:34: warning: 'Atomic_AND_u32' defined but not used [-Wunused-function]
 static portFORCE_INLINE uint32_t Atomic_AND_u32(
                                  ^
../freertos_kernel/include/atomic.h:369:34: warning: 'Atomic_NAND_u32' defined but not used [-Wunused-function]
 static portFORCE_INLINE uint32_t Atomic_NAND_u32(
                                  ^
../freertos_kernel/include/atomic.h:397:34: warning: 'Atomic_XOR_u32' defined but not used [-Wunused-function]
 static portFORCE_INLINE uint32_t Atomic_XOR_u32(
                                  ^
```

Can you please check if you observe such warnings in case of other boards as well?